py/sentry_streams/user_functions/function_template.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class Accumulator(ABC):
```

Comment:
How is this going to be related with the storage choice? 
Are Accumulator and Storage separate entities or would they be connected ?

Reply:
Added `AggregationBackend`. It's a WIP that needs some feedback.